ci/test_wheel_cudf.sh

```diff
@@ -3,9 +3,13 @@
 
 set -eou pipefail
 
+# Download the pylibcudf built in the previous step
 RAPIDS_PY_CUDA_SUFFIX="$(rapids-wheel-ctk-name-gen ${RAPIDS_CUDA_VERSION})"
-RAPIDS_PY_WHEEL_NAME="cudf_${RAPIDS_PY_CUDA_SUFFIX}" rapids-download-wheels-from-s3 ./dist
+RAPIDS_PY_WHEEL_NAME="pylibcudf_${RAPIDS_PY_CUDA_SUFFIX}" rapids-download-wheels-from-s3 ./local-pylibcudf-dep
+python -m pip install ./local-pylibcudf-dep/pylibcudf*.whl
 
+# Install cudf
+RAPIDS_PY_WHEEL_NAME="cudf_${RAPIDS_PY_CUDA_SUFFIX}" rapids-download-wheels-from-s3 ./dist
 # echo to expand wildcard before adding `[extra]` requires for pip
 python -m pip install $(echo ./dist/cudf*.whl)[test]
```

Comment:
I think it would be a bit safer to combine the two `pip install` calls, like this:

```shell
python -m pip install \
    $(echo ./dist/cudf*.whl)[test] \
    ./local-pylibcudf-dep/pylibcudf*.whl
```

That way, you can be certain that `pip install` of `cudf` doesn't lead to a new `pylibcudf` being pulled in from a remote index (overwriting the one you just downloaded).